### PR TITLE
Refine ServiceDiscovery in Serverable API

### PR DESCRIPTION
**Pull request by yaooqinn**
yaooqinn [Closes #453](https://github.com/yaooqinn/kyuubi/pull/453)    Target Issue Feature [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=yaooqinn&utm_campaign=badge_info)

### _Why are the changes needed?_

A service discovery should be a default module of Severable API and shall be initialized after `FroundendService` for a resolved serviceUri.

it's a reflection only fix to enhance this logic

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/latest/tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: kyuubi-common/src/test/scala/org/apache/kyuubi/service/NoopServer.scala [hunk 1]
```diff
@@ -32,4 +32,8 @@ class NoopServer extends Serverable("noop") {
   override protected def stopServer(): Unit = {
     throw new KyuubiException("no need to stop me")
   }
+
+
+  override protected val discoveryService: Service = backendService
```
Comment (yaooqinn) [inline kyuubi-common/src/test/scala/org/apache/kyuubi/service/NoopServer.scala:37]: this is just a hack and just be fine

File: kyuubi-main/src/main/scala/org/apache/kyuubi/server/KyuubiServer.scala [hunk 2]
```diff
@@ -91,11 +94,8 @@ class KyuubiServer(name: String) extends Serverable(name) {
     }
 
     super.initialize(conf)
-    if (ServiceDiscovery.supportServiceDiscovery(conf)) {
```
Comment (yaooqinn) [inline kyuubi-main/src/main/scala/org/apache/kyuubi/server/KyuubiServer.scala:94]: all services should be added ahead and init/start in super.init/start

**Comment by yaooqinn** [comment]
cc @pan3793 @ulysses-you

**Review by ulysses-you — APPROVED** [review]

**Comment by yaooqinn** [comment]
thanks, merged to master for v1.2.0
